File: assignment3/python/furniture.py
class Furniture:
    def __init__(self, material : str):
        self._material : str = material
        self._sit = False
    
    @property
    def material(self) -> str:
        return self._material

    @property
    def sit(self) -> bool:
        return self._sit
    
    def sit(self) -> None:
        if self._sit == True:
            raise ValueError("Cannot sit! Already sitting.")
        self._sit = True
        
    def stand(self) -> None:
        self._sit = False

File: assignment3/c++/src/furniture.cpp
#include <stdexcept>
#include "furniture.h"

Furniture::Furniture(std::string material) {
    _material = material;
    _sit = false;
}

std::string Furniture::getMaterial() const { return _material; }
bool Furniture::alreadySitting() const { return _sit; }

void Furniture::sitDown() {
    if(_sit)
        throw std::invalid_argument("Cannot sit! Already sitting.");
    _sit = true;
}

void Furniture::standUp() {
    _sit = false;
}